Source/JavaScriptCore/API/JSBase.h

```diff
@@ -68,6 +68,9 @@ typedef const struct OpaqueJSValue* JSValueRef;
 /*! @typedef JSObjectRef A JavaScript object. A JSObject is a JSValue. */
 typedef struct OpaqueJSValue* JSObjectRef;
 
+/*! @typedef JSBigIntRef A JavaScript bigint. A JSBigInt is a JSValue. */
+typedef const struct OpaqueJSValue* JSBigIntRef;
```

Comment:
I guess I don't really get the point of having a JSBigIntRef type. You can't do anything with it that you can't do with the JSValue. We don't really have special types for other primitive types with the exception of JSStringRef, which gives you back something other than a JSValue anyway. Can you explain your thought process here?

Reply:
I think there are several reasons:
1. Unlike other primitives (except for JSString) which can be directly converted to C types.
2. With JSBigIntRef, no need to check whether it's bigint or not for its related APIs (JSBigIntToInt64, JSBigIntToUInt64, .etc).
3. JSBigIntRef follows modular API design which breaks down JSValue into smaller and self-contained components which can be developed, deployed, and maintained independently. Bigint arithmetic operations, which requires APIs due to (1), is not introduced yet but might be added on demands. If those operations are needed, then JSBigIntRef is beneficial since (2).

Reply:
I don't think I agree with your arguments here. As this API stands right now, JSBigIntRef adds a lot of surface area but I don't see how it makes life more convenient for users of the API. If we decide to add arithmetic operations in the future we can always add JSBigIntRef at that point. I'm also not convinced that eliminating the checks is a significant win. Users already have to call a function to do the addition, which likely involves an object allocation. Lastly, there are primitives that don't have a C type other than JSString, namely JSSymbol, which don't have a dedicated type. 

IMO, if we want to have a more modular API we should probably do it in the Obj-C/Swift API first since that API is more widely used and it actually has proper subclassing. Also, we should do it more comprehensively with many subclasses for many different types.